### DISCUSSION TITLE
Add paginator information to API results

### DISCUSSION THREAD

**Discussion by idflood** [directus/directus#5348] [Feature Requests]
Hi,

When consuming data to create a website, with a list of articles for instance, having access to pagination data is nearly required. Getting all articles to "manually" count them is not great, especially if the blog has a few hundreds articles.

I don't know if it's still possible to do breaking change to API since it's a RC realease, but I would suggest to change the GraphQL result to something like this:

```
query {
  article(filter: {...}) {
    data {
      title
      ...
    }
    paginator {
      total
      page
      ...
    }
  }
}
```

This would allow to get all the informations needed to build an article listing with pagination in one query. Also the paginator data should take into account the possible filters we give to the query.

For REST result the paginator information could be returned in headers, or following the same sturcture as the REST request.

**Comment by rijkvanzanten** [comment]
The REST API already has a `meta` query param that can be used to access this sort of information(https://docs.directus.io/reference/api/query/#metadata). There's a discussion over here on how to structure that ideally under GraphQL: https://github.com/directus/directus/discussions/4636

FWIW, we used to item data under a nested key similar to your example, but that was causing issues in GraphQL libraries that do their own caching, like apollo/relay, so that was flattened

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
